### PR TITLE
RIA-5626 Hearing Centre mappings, entity, service and handler

### PR DESCRIPTION
**Pull request by Ryan-Nesbitt**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RIA-5626

### Change description ###

Added YML file for hearing centre mappings, linked into application.yml.
Added hearingCentre CaseField into the BailCaseFieldDefinitions.
Added entity for HearingCentre, and unit tests.
Added service for HearingCentreFinder and unit tests.
Added handler for DeriveHearingCentreHandler and unit tests.
Added configuration for HearingCentreFinderConfiguration.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/DeriveHearingCentreHandler.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        return new PreSubmitCallbackResponse<>(bailCase);
+    }
+
+    public void setHearingCentreFromDetentionFacilityName(BailCase bailCase) {
+        final String prisonName = bailCase.read(PRISON_NAME, String.class).orElse("");
+
+        final String ircName = bailCase.read(IRC_NAME, String.class).orElse("");
+
+        String detentionFacilityName = !prisonName.isEmpty() ? prisonName : ircName;
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/DeriveHearingCentreHandler.java:56]: Can you put a check here - if both prison name and irc name are empty, it should throw RequiredFieldMissingException.

**Review by neha-aggarw — CHANGES_REQUESTED** [review]
Can you please add the check as mentioned.

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/DeriveHearingCentreHandler.java (hunk 1 above)
Comment (Ryan-Nesbitt) [reply src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/DeriveHearingCentreHandler.java]: Added in a check and a test for the exception. Have a look and see if it's alright

**Review by neha-aggarw — APPROVED** [review]
LGTM

**Review by Dbarnes-v1 — APPROVED** [review]
LGTM

**Pull request merged**